runtime/lua/vim/lsp.lua

```diff
@@ -16,24 +7,39 @@ local tbl_isempty, tbl_extend = vim.tbl_isempty, vim.tbl_extend
 local validate = vim.validate
 local if_nil = vim.F.if_nil
 
-local lsp = {
-  protocol = protocol,
-
-  handlers = default_handlers,
+local submodules = {
+  '_changetracking',
+  '_completion',
+  '_dynamic',
+  'buf',
+  'codelens',
+  'diagnostic',
+  'handlers',
+  'inlay_hint',
+  'log',
+  'protocol',
+  'rpc',
+  'semantic_tokens',
+  'tagfunc',
+  'util',
+}
+local dundarload_submodules = vim._dundarload_submodules('vim.lsp', submodules)
+local lsp = setmetatable({}, dundarload_submodules)
 
-  buf = require('vim.lsp.buf'),
-  diagnostic = require('vim.lsp.diagnostic'),
-  codelens = require('vim.lsp.codelens'),
-  inlay_hint = require('vim.lsp.inlay_hint'),
-  semantic_tokens = semantic_tokens,
-  util = util,
+local default_handlers = lsp.handlers
+local log = lsp.log
+local lsp_rpc = lsp.rpc
+local protocol = lsp.protocol
+local ms = protocol.Methods
+local util = lsp.util
+local changetracking = lsp._changetracking
+local semantic_tokens = lsp.semantic_tokens
```

Comment:
this block negates most of the effects of `_dunderload_submodules`. Is that intentional?

Reply:
Huh, you're absolutely right. I missed that. Thanks.

Reply:
I've kept some of the more regular ones. Lemme know if you think I should dundarfy them as well.